joeynmt/training.py

```diff
@@ -65,8 +65,10 @@ def __init__(self, model: Model, config: dict) -> None:
 
         self.logging_freq = train_config.get("logging_freq", 100)
         self.valid_report_file = "{}/validations.txt".format(self.model_dir)
-        self.tb_writer = SummaryWriter(
-            log_dir=self.model_dir + "/tensorboard/")
+        self.tb_writer = SummaryWriter(log_dir=self.model_dir + "/tensorboard/")
+
+        self.save_latest_checkpoint = train_config.get("save_latest_ckpt",
+                                                       False)
```

Comment:
I think we can safely default to True, or does this cause any issues with backward compatibility of existing models?

Reply:
This shouldn't cause any backwards compatibility problems, my concern was that it saves a new checkpoint every validation and then deletes the previous one, which could be a lot of read/writes. I have to ask if that would be bad for SSDs at all.

Reply:
Have you tried this in a training run how much of an overhead it is?

Reply:
I'll start an iwslt 'groundhog' model and compare run times to see what overhead this introduces. I asked hiko and he said it would take years before this would affect ssd performance if you write a 2gb model once an hour.

Reply:
ok good :D I trust Hiko's estimate.

Reply:
Training 18000 steps took 58 minutes, 7 seconds without saving the latest checkpoint each time while saving the latest checkpoint took 59 minutes and 8 seconds. That was validating every 2000 steps, so it saved the latest checkpoint 9 times, It's also worth noting that in both runs every validation was better than the last, so the best.ckpt was being updated every time as well.

Reply:
that sounds fine, thanks for running the test!